pkg/clients/certificatemanager/certificate.go

```diff
@@ -0,0 +1,173 @@
+package certificatemanager
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/service/acm"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+
+	"github.com/crossplane/provider-aws/apis/certificatemanager/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// Client defines the CertificateManager operations
+type Client interface {
+	// GetCertificateRequest(*acm.GetCertificateInput) acm.GetCertificateRequest
+	DescribeCertificateRequest(*acm.DescribeCertificateInput) acm.DescribeCertificateRequest
+	RequestCertificateRequest(*acm.RequestCertificateInput) acm.RequestCertificateRequest
+	DeleteCertificateRequest(*acm.DeleteCertificateInput) acm.DeleteCertificateRequest
+	UpdateCertificateOptionsRequest(*acm.UpdateCertificateOptionsInput) acm.UpdateCertificateOptionsRequest
+	ListTagsForCertificateRequest(*acm.ListTagsForCertificateInput) acm.ListTagsForCertificateRequest
+	AddTagsToCertificateRequest(*acm.AddTagsToCertificateInput) acm.AddTagsToCertificateRequest
+	RenewCertificateRequest(*acm.RenewCertificateInput) acm.RenewCertificateRequest
+	RemoveTagsFromCertificateRequest(*acm.RemoveTagsFromCertificateInput) acm.RemoveTagsFromCertificateRequest
+}
+
+// NewClient returns a new client using AWS credentials as JSON encoded data.
+func NewClient(conf *aws.Config) (Client, error) {
+	return acm.New(*conf), nil
+}
+
+// GenerateCreateCertificateInput from CertificateSpec
+func GenerateCreateCertificateInput(name string, p *v1alpha1.CertificateParameters) *acm.RequestCertificateInput {
+	m := &acm.RequestCertificateInput{
+		DomainName:              aws.String(p.DomainName),
+		CertificateAuthorityArn: p.CertificateAuthorityArn,
+		IdempotencyToken:        p.IdempotencyToken,
+	}
+
+	if strings.EqualFold(p.CertificateTransparencyLoggingPreference, "DISABLED") {
+		m.Options = &acm.CertificateOptions{CertificateTransparencyLoggingPreference: acm.CertificateTransparencyLoggingPreferenceDisabled}
+	} else if strings.EqualFold(p.CertificateTransparencyLoggingPreference, "ENABLED") {
+		m.Options = &acm.CertificateOptions{CertificateTransparencyLoggingPreference: acm.CertificateTransparencyLoggingPreferenceEnabled}
+	}
+
+	if strings.EqualFold(p.ValidationMethod, "EMAIL") {
+		m.ValidationMethod = acm.ValidationMethodEmail
+	} else if strings.EqualFold(p.ValidationMethod, "DNS") {
+		m.ValidationMethod = acm.ValidationMethodDns
+	}
```

Comment:
Would be good to make these `enum` types on the CRD, ref: https://github.com/crossplane/provider-aws/blob/d2ddf8ff9521dcbfbeba2215f35f9b401414e1ef/apis/network/v1alpha3/internetgateway_types.go#L52

Reply:
Thanks, Changed the type to Enum.